src/main/java/gift/constant/KakaoClientConfig.java

```diff
@@ -0,0 +1,46 @@
+package gift.constant;
+
+import gift.kakao.KakaoMessageInterface;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.web.client.RestClient;
+import org.springframework.web.client.support.RestClientAdapter;
+import org.springframework.web.service.invoker.HttpServiceProxyFactory;
+
+@Configuration
+public class KakaoClientConfig {
+
+    @Value("${custom.kakao-auth-server}")
+    private String kakao_auth_server;
+
+    @Value("${custom.kakao-sendTo-myself}")
+    private String kakao_self_message;
+
+    @Bean
+    public RestClient kakaoAuthRequestClient() {
+        return RestClient.builder()
+            .baseUrl(kakao_auth_server)
+            .defaultHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON_VALUE)
+            .build();
+    }
+
+    @Bean
+    public RestClient.Builder kakaoMessageRequestClient() {
+        return RestClient.builder()
+            .baseUrl(kakao_self_message)
+            .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED_VALUE);
+    }
+
+    @Bean
+    public KakaoMessageInterface createKakaoMessageClient(RestClient kakaoRestClient) {
+
+        return HttpServiceProxyFactory
+            .builder()
+            .exchangeAdapter(RestClientAdapter.create(kakaoRestClient))
+            .build()
+            .createClient(KakaoMessageInterface.class);
+    }
```

Comment:
```suggestion
    @Bean
    public KakaoMessageInterface createKakaoMessageClient() {
        RestClient restClient = RestClient.builder()
                                          .baseUrl(kakao_self_message)
                                          .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED_VALUE)
                                          .build();

        RestClientAdapter adapter = RestClientAdapter.create(restClient);
        HttpServiceProxyFactory factory = HttpServiceProxyFactory.builderFor(adapter).build();

        return factory.createClient(KakaoMessageInterface.class);
    }
```

이런식으로 하면 좀 더 깔끔하겠네요~